### ISSUE TITLE
Not all input data has been trained

### ISSUE THREAD

**Issue by praveenkreddy** [LossNAN/I3D-Tensorflow#22]
From the code below it seems that at each step only the number of videos trained at each step is equal to batch size and also these videos are chosen at random as i see next_batch_start value from the input_data.py is not fed back in the next step.
 for step in xrange(FLAGS.max_steps):
        start_time = time.time()
        rgb_train_images, flow_train_images, train_labels, _, _, _ = input_data.read_clip_and_label(
                      filename='../../list/ucf_list/train.txt',
                      batch_size=FLAGS.batch_size * gpu_num,
                      num_frames_per_clip=FLAGS.num_frame_per_clib,
                      crop_size=FLAGS.crop_size,
                      shuffle=True
                      )
To my understanding all the training set should be trained at each step. Can the @author or anyone who is successfull in implementing this repo please help me understand how this is working.

**Comment by ilkarman** [comment]
A step in tensorflow is a forward (+ back) pass; whereas an epoch is however many steps (depending on batch-size) to get through all your data once. The next_btach_start value is used during testing where the whole clip is loaded rather than sampling.

**Comment by bks-abs** [comment]
i  have the same question. can anyone help me clear~~

**Comment by huangchaoxing** [comment]
All the training set should be trained in one epoch. Every-step is just a batch.

**Comment by YuLengChuanJiang** [comment]
@ilkarman hello,  have you reach the accuarcy that  descirbed in the repo on the UCF101 using i3d,  I can only reach 0.89 for RGB on split 1.
